primitives/runtime/src/offchain/storage.rs

```diff
@@ -62,11 +91,21 @@ impl<'a> StorageValueRef<'a> {
 	///
 	/// The function returns `None` if the value was not found in storage,
 	/// otherwise a decoding of the value to requested type.
-	pub fn get<T: codec::Decode>(&self) -> Option<Option<T>> {
-		sp_io::offchain::local_storage_get(self.kind, self.key)
-			.map(|val| T::decode(&mut &*val).ok())
+	pub fn get<T: codec::Decode>(&self) -> StorageEntry<T> {
+		if let Some(val) = sp_io::offchain::local_storage_get(self.kind, self.key)
```

Comment:
If `get` returns `Result<T, StorageEntryError>`, then we could simplify by writing this line as 

```suggestion
		let val = sp_io::offchain::local_storage_get(self.kind, self.key).ok_or(StorageEntryError::KeyNotStored)?;
```